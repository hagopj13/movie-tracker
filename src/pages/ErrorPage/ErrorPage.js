// @flow
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';

import brokenPage from './brokenPage.png';

type Props = {
  errorText: string,
};

const useStyles = makeStyles((theme) => ({
  root: {
    height: '80vh',
    width: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  image: {
    width: '40vh',
    height: '40vh',
    marginBottom: theme.spacing(3),
  },
  errorText: {
    fontWeight: theme.typography.fontWeightMedium,
    textAlign: 'center',
  },
}));

const ErrorPage = (props: Props) => {
  const { errorText } = props;

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <img className={classes.image} src={brokenPage} alt="page broken" />
      <Typography className={classes.errorText} color="primary" variant="h3">
        {errorText}
      </Typography>
    </div>
  );
};

export default ErrorPage;
