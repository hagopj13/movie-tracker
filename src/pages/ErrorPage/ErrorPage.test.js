import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import Typography from '@material-ui/core/Typography';

import ErrorPage from './ErrorPage';

describe('ErrorPage page', () => {
  let shallow;
  let errorText;
  let wrapper;

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    errorText = 'Some error text';
    const mockProps = {
      errorText,
    };
    wrapper = shallow(<ErrorPage {...mockProps} />);
  });

  it('should render the ErrorPage page correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the error text', () => {
    expect(wrapper.find(Typography).text()).toBe(errorText);
  });
});
