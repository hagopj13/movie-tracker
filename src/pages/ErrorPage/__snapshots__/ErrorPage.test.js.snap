// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage page should render the ErrorPage page correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <img
    alt="page broken"
    className="makeStyles-image-2"
    src="brokenPage.png"
  />
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-errorText-3"
    color="primary"
    variant="h3"
  >
    Some error text
  </WithStyles(ForwardRef(Typography))>
</div>
`;
